frontend/src/views/Lobby.tsx

```diff
@@ -65,35 +66,37 @@ function LobbyPage() {
 
   // Grab the nickname variable and add the user to the lobby.
   useEffect(() => {
-    // Grab the nickname variable; otherwise, set an error.
-    if (location && location.state && location.state.nickname) {
-      setNickname(location.state.nickname);
+    // Grab the user and room information; otherwise, redirect to the join page
+    if (location && location.state && location.state.user && location.state.room) {
+      setCurrentUser(location.state.user);
+      setRoom(location.state.room);
     } else {
-      setError('No nickname was provided for the user in the lobby.');
+      setShouldRedirect(true);
     }
 
     // Connect the user to the room.
-    if (!socketConnected && nickname) {
-      connectUserToRoom(socketRoomId, nickname);
+    if (!socketConnected && room !== null) {
+      connectUserToRoom(room.roomId);
     }
-  }, [location, socketConnected, nickname, connectUserToRoom]);
+  }, [location, socketConnected, room, connectUserToRoom]);
 
   // Render the lobby.
   return (
     <div>
+      { shouldRedirect ? <Redirect to="/game/join" /> : null}
       <LargeText>
-        You have entered the lobby! Your nickname is &quot;
-        {nickname}
+        You have entered the lobby for room
+        {' '}
+        {room?.roomId}
+        ! Your nickname is &quot;
+        {currentUser?.nickname}
         &quot;.
       </LargeText>
       { error ? <ErrorMessage message={error} /> : null }
       <div>
         {
-          users.map((user) => (
-            <UserNicknameText onClick={(event) => {
-              deleteUser((event.target as HTMLElement).innerText);
-            }}
-            >
+          room?.users.map((user) => (
+            <UserNicknameText onClick={() => deleteUser(user)}>
```

Comment:
This is probably better for a later PR -- should we have a pop up message or something asking the user to confirm they want to delete someone?

Reply:
Yeah, right now it just does a simple `console.log` since we don't have that functionality yet. Lmk if you think I should change that to a pop-up or something tho